notes/css.md

```diff
@@ -0,0 +1,248 @@
+:is(article, aside, section, nav) + :is(article, aside, section, nav) {
+	margin-block-start: 1rem;
+}
+```
+
+Це аналогічно до перебору усіх комбінацій:
+```css
+article + article,
+article + aside,
+article + section,
+article + nav,
+
+aside + article,
+/* ... 11 селекторів опущено */
+nav + nav {
+	margin-block-start: 1rem;
+}
+```
+
+`:where()` працює аналогічно, але має специфічність 0.
+
+
+### `:has()`
+
+Дозволяє стилізувати елемент, який має певні дочірні або сусідні елементи, які відповідають селектору всередині.
+
+`article:has(h2)` — це така стаття, всередині якої є заголовок другого рівня.
+
+
+### Псевдо-елементи
+
+`::before` та `::after` для того, щоб можна було дадавати _презентаційний_ контент елементам —
+такий, що не несе смислового навантаження, ні візуального, ні текстового.
+Якщо без контенту в них не можна обійтися, або треба, щоб він копіювався,
+цей контент має бути в розмітці, а не в стилях.
+
+`::before` та `::after` не можна виділити чи скопіювати, їх немає в [replaced elements](https://developer.mozilla.org/docs/Web/CSS/Replaced_element) (`img`, `input`); вони додаються тільки коли є `content`, і стандартно мають `display: inline`.
+
+Інші псевдо-елементи додволяють стилізувати те, чого явно немає в дереві документу —
+виділення (`::selection`),
+маркери списків (`::marker`),
+текст-підказка (`::placeholder`),
+першу літеру чи рядок (`::first-letter`, `::first-line`).
+
+[Список псевдо-елементів на MDN](https://developer.mozilla.org/docs/Web/CSS/Pseudo-elements)
+
+### Псевдо-класи
+
+Для того, щоб можна було доступитися до специфісних станів елементів, наприклад, `:visited`, `:invalid`.
+
+[Список псевдо-класів на MDN](https://developer.mozilla.org/docs/Web/CSS/Pseudo-classes)
+
+
+Values and Units
+-
+Використовуйте `rem` та `em` для віступів та полів (`margin`, `padding`, `gap` тощо).
+Використовуйте ці та інші відносні до розміру шрифту величини — `ch`, `lh`, `ex`, `cap`, а також відсотки (`%`) для встановлення розмірів елементів, де це необхідно.
+Це робить дизайн гнучким до налаштувань шрифту **користувача**.
+
+Використовуйте <b>новий синтаксис</b> для написання кольорів:
+```
+Старий:
+rgba(r, g, b)
+rgb(r, g, b)
+
+Новий:
+rgb(r g b / a)
+hsl(h s l / a)
+```
+
+Генерувати кольорові схеми <mark>з логічною градацією світності</mark> найзручніше в просторах LAB та OKLAB:
+```
+lch(lighness chroma hue / alpha)
+oklch(lighness chroma hue / alpha)
+```
+
+Список на [MDN](https://developer.mozilla.org/docs/Learn/CSS/Building_blocks/Values_and_units)
+Специфікація про кольори: [CSS Color Module Level 4](https://www.w3.org/TR/css-color-4/)
+
+
+Box Model
+-
+
+![CSS Box Model](https://www.w3.org/TR/css-box-4/images/box.png)
+
+Специфікація: [CSS Box Model Level 4](https://www.w3.org/TR/css-box-4/)
+
+
+### Display
+
+<b>Відображення</b> елемента є:
+- <b>зовнішнє</b>&nbsp;— те, як він веде себе в просторі батькіського елемента; та
+- <b>внутрішнє</b>&nbsp;— те, яке середовище він створює для дочірніх елементів.
```

Comment:
```suggestion
**Відображення** елемента є:
- **зовнішнє**&nbsp;— те, як він веде себе в просторі батькіського елемента; та
- **внутрішнє**&nbsp;— те, яке середовище він створює для дочірніх елементів.
```

Reply:
strong = importance
b = keyword highlighting

I highlighted keywords but I don't find this as more important than the rest of text.

Reply:
@viktor-yakubiv my point is don't use html tags for styling in markdown. Markdown has its own mechanism for it. In stage 0 we told them about it, but in Stage 1  we are does the opposite.. :grin:

Reply:
Why? For the syntax sanity?

Markdown is a superset of HTML which means any HTML is normal in Markdown. When Markdown cannot achieve something with its limited syntax, HTML comes for the rescue.

The GitHub's point of view onto Markdown is that HTML may be unsafe, therefore they strip most of the elements from any Markdown text. If am fine with that. Although, I prefer `<b>` over `<strong>` (aka `__` or `**` syntax) as this is more **correct** and less inaccessible way of writing text.

I find forbidding HTML in Markdown similar to forbidding tag omitting in HTML because of missing such a feature in XML, XHTML, therefor JSX. I honestly don't understand why you taught students to avoid HTML in Markdown; it would be kind if you could elaborate. Thank you 🙂